src/App.tsx

```diff
@@ -14,11 +17,160 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  initialGoods: string[];
+  isStarted: boolean;
+  isReversed: boolean;
+  sortBy: string;
+  minLength: number;
+}
+
+class App extends React.PureComponent<{}, State> {
+  state: State = {
+    initialGoods: goodsFromServer,
+    isStarted: true,
+    isReversed: false,
+    sortBy: 'reset',
+    minLength: 1,
+  };
+
+  onStartButtonClick = () => {
```

Comment:
This naming is suitable for a component props, not for handler functions

Read more: https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364